### ISSUE TITLE
cred.credentials: implements -> implementer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7620]
|<img alt="wolfgang61's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| wolfgang61 reported|
|-|-|
|Trac ID|trac#7620|
|Type|enhancement|
|Created|2014-09-02 08:09:40Z|
|Branch|https://github.com/twisted/twisted/tree/cred-implements-implementer-7620-2|

preparing for python 3

Attachments:

* [credimplpatch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4f1/4f12aafbf75e6a7775da73abcf3ab39cdd0dfd00/8a4fd149fe859fb2d77b1d047e970d2c4ee40b90) (2433 bytes) - added by wolfgang61 on 2014-09-02 08:10:35Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7620 7620
type__enhancement enhancement
reporter__wolfgang61 wolfgang61
priority__normal normal
milestone__ 
branch__branches_cred_implements_implementer_7620_2 branches/cred-implements-implementer-7620-2
branch_author__jml jml
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1409645380086332 1409645380086332
changetime__1474198049187875 1474198049187875
version__None None
owner__jml jml

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml set owner to wolfgang61|
|-|-|

Hello Wolfgang,

Thanks for the patch. Could you please:

* add some context to this ticket explaining _why_ this patch is a good idea
* provide a news file, as per https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles – I think it can be a 'misc' one in this case

Once you've done those things, please resubmit for review and assign back to me.

Thanks again,
jml

**Comment by twisted-trac** [comment]
|<img alt="wolfgang61's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|wolfgang61 set owner to @jml|
|-|-|

Replying to [jml](#note_1):
> Thanks for the patch. Could you please:
> 
> * add some context to this ticket explaining _why_ this patch is a good idea

Not sure where you want this. Right here? So – This is needed because zope.interface deprecated "implements" and introduced "@implementer". With zope.interface for Python 3, "implements" is not available anymore.

> * provide a news file, as per https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles – I think it can be a 'misc' one in this case

I was waiting for ticket 7628 to be resolved. That would introduce 'port' for news files. But since my Ticket 7621 with the same change is already in trunk as 'misc' and since this is minor even for a porting change, I now added 7620.misc.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

(In [43368]) Branching to cred-implements-implementer-7620.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml commented|
|-|-|

(In [43381]) Branching to cred-implements-implementer-7620-2.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [43387]) Merge cred-implements-implementer-7620-2: Replace implements with implementer in t.cred.credentials

Author: wolfgang61
Reviewer: jml
Fixes: [#7620](https://github.com/twisted/twisted/issues/7620)

The @implementer class decorator is a Python 3-safe way of declaring
that a class implements an interface.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
